lib/openshift/project.rb

```diff
@@ -104,6 +104,13 @@ def create(by:, **opts)
       else
         res = by.cli_exec(:new_project, project_name: name, **opts)
       end
+      success = wait_for(60, interval: 5) {
+        uid_range = by.cli_exec(:get, resource: "namespace", resource_name: name, template: '{{ index .metadata.annotations "openshift.io/sa.scc.uid-range" }}')
+        !uid_range.nil? && !uid_range.empty? && uid_range !~ /no value/
+      }
+      unless success
+        raise "timeout waiting for project #{name} to get annotation openshift.io/sa.scc.uid-range"
```

Comment:
Given applying wider to every project creation run by every case, if timeout, still not easy to debug whether it is a bug or flaky, debugger does not know whether it needs a few more time, or it never appears. Suggest to increase 60 to bigger for example 120 or more, but record the actual time T it uses, and then assert unless (success or T < 60). Like the method used here https://github.com/openshift/cucushift/pull/8911/files .
And if timeout happens even if increased to bigger, it is better if we can dump some logs, not sure, this needs @kasturinarra to give some insights, seems logs from kube-controller-manager cluster-policy-controller container, can only grep project name from the logs.

Reply:
This is more like a workaround to let the CI move on, rather than a fix, since the issue comes from OCP side.

IMO, we should initially start with a smaller timeout (e.g, 60 seconds, maybe 30 seconds or even less, as generally the project get the annotation in a few seconds) here. Then we observe how it behave from the log history. And then decide whether we should report a bug to OCP, or increase the timeout a little bit.

It's really a bad user experience if a user have to wait for minutes or more to create their pods after the creation of their project.

Reply:
Assertion "unless (T < 60)" does not let the CI move on, it is to help debug when it happend to fail due to have to wait for minutes or more to create their pods after the creation of their project

Reply:
`wait_for` means wait up to xx seconds at max. So if you got a success, then the wait time is less or equal to xx seconds. But if you got a fail, then the wait time is larger than xx seconds.

Reply:
> But if you got a fail, then the wait time is larger than xx seconds.

This is expected by the design, which intends to help debug by telling the debugger exactly "whether it needs a few more time, or it never appears" when you got a failure.

Reply:
I remember seems there was a bug raised but no data then dev closed? I can't find the bug. But seems the issue is sometimes often hit in CI. If not more data shown in CI output, Dev likely wants more data from us again and close again?

Reply:
`timeout waiting for project prj-7e7581d21d83-rootx-49im to get annotation openshift.io/sa.scc.uid-range after 62.102225416 seconds, consider reporting a bug (RuntimeError)`, is it what you would expect it to ?

Reply:
Thank you for working. Sorry for disturbing weekend enjoying if any :)
Yes it looks good. But "    unless success" should be changed to "    unless wait_since <= 60"
And another point, not sure if using using `stats: stats` and stats[:seconds] lke https://github.com/openshift/verification-tests/blob/master/features/step_definitions/operators.rb#L194 is preferred to get the waiting time.
/hold

Reply:
/unhold

Reply:
No other issue from my side. Thanks